pkg/auth/token.go

```diff
@@ -109,3 +111,7 @@ func Token(
 
 	return signedtoken, nil
 }
+
+func IsPublic(ctx context.Context) bool {
+	return metautils.ExtractIncoming(ctx).Get("authorization") == ""
```

Comment:
Use a constant for the string and investigate that the key is case-insensitive

Reply:
it does toLower: https://github.com/grpc-ecosystem/go-grpc-middleware/blob/master/util/metautils/single_key.go#L16